api/src/main/java/org/pmiops/workbench/opsgenie/EgressEventServiceImpl.java

```diff
@@ -52,22 +55,36 @@ public EgressEventServiceImpl(
       Provider<AlertApi> alertApiProvider,
       Provider<WorkbenchConfig> workbenchConfigProvider,
       UserService userService,
-      WorkspaceAdminService workspaceAdminService) {
+      WorkspaceAdminService workspaceAdminService,
+      WorkspaceDao workspaceDao) {
     this.clock = clock;
     this.egressEventAuditor = egressEventAuditor;
     this.institutionService = institutionService;
     this.alertApiProvider = alertApiProvider;
     this.workbenchConfigProvider = workbenchConfigProvider;
     this.userService = userService;
     this.workspaceAdminService = workspaceAdminService;
+    this.workspaceDao = workspaceDao;
   }
 
   @Override
   public void handleEvent(EgressEvent event) {
+    // Lookup workspace by googleProject name, and set the workspaceNamespace in EgressEvent.
+    String workspaceNamespace =
+        workspaceDao
+            .getByGoogleProject(event.getProjectName())
+            .orElseThrow(
+                () ->
+                    new NotFoundException(
+                        String.format(
+                            "Workspace not found by given Google Project Id: %s",
+                            event.getProjectName())))
+            .getWorkspaceNamespace();
+    event.setWorkspaceNamespace(workspaceNamespace);
     logger.warning(
         String.format(
-            "Received an egress event from project %s (%.2fMiB, VM prefix %s)",
-            event.getProjectName(), event.getEgressMib(), event.getVmPrefix()));
+            "Received an egress event from workspace namespace %s (%.2fMiB, VM prefix %s)",
```

Comment:
would be useful to have both google project ID and workspace namespace here, I think